File: Data Structures & Algorithms/house-robber/submission-23.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    rob(nums) { 
        let prev1 = 0 
        let prev2 = 0 

        for (let i = 0; i < nums.length; i ++) {
            let cur = Math.max(nums[i] + prev1, prev2) 
            let temp = prev2 
            prev2 = cur 
            prev1 = temp
        } 

        return prev2
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-23.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums) 

        res = 0 

        for num in numSet: 
            if num - 1 in numSet: 
                continue 

            cur = num  
            cnt = 0
            while cur in numSet: 
                cur += 1 
                cnt += 1 

            res = max(res, cnt) 

        return res